lib/compat/wordpress-6.1/class-wp-theme-json-6-1.php

```diff
@@ -214,6 +360,63 @@ private static function get_block_nodes( $theme_json, $selectors = array() ) {
 		return $nodes;
 	}
 
+	/**
+	 * Returns the stylesheet that results of processing
+	 * the theme.json structure this object represents.
+	 *
+	 * @param array $types    Types of styles to load. Will load all by default. It accepts:
+	 *                         'variables': only the CSS Custom Properties for presets & custom ones.
+	 *                         'styles': only the styles section in theme.json.
+	 *                         'presets': only the classes for the presets.
+	 * @param array $origins A list of origins to include. By default it includes VALID_ORIGINS.
+	 * @return string Stylesheet.
+	 */
+	public function get_stylesheet( $types = array( 'variables', 'styles', 'presets' ), $origins = null ) {
+		if ( null === $origins ) {
+			$origins = static::VALID_ORIGINS;
+		}
+
+		if ( is_string( $types ) ) {
+			// Dispatch error and map old arguments to new ones.
+			_deprecated_argument( __FUNCTION__, '5.9' );
+			if ( 'block_styles' === $types ) {
+				$types = array( 'styles', 'presets' );
+			} elseif ( 'css_variables' === $types ) {
+				$types = array( 'variables' );
+			} else {
+				$types = array( 'variables', 'styles', 'presets' );
+			}
+		}
+
+		$blocks_metadata = static::get_blocks_metadata();
+		$style_nodes     = static::get_style_nodes( $this->theme_json, $blocks_metadata );
+		$setting_nodes   = static::get_setting_nodes( $this->theme_json, $blocks_metadata );
+
+		$stylesheet = '';
+
+		if ( in_array( 'variables', $types, true ) ) {
+			$stylesheet .= $this->get_css_variables( $setting_nodes, $origins );
+		}
+
+		if ( in_array( 'styles', $types, true ) ) {
+			$stylesheet .= $this->get_block_classes( $style_nodes );
+		} elseif ( in_array( 'block-layout-styles', $types, true ) ) {
+			$stylesheet .= $this->get_layout_styles(
+				array(
+					'path'     => array( 'styles' ),
+					'selector' => static::ROOT_BLOCK_SELECTOR,
+				),
+				true
+			);
+		}
```

Comment:
The only change from `6.0` in this function is this set of lines, which will conditionally output just those layout styles required for blocks (only if `styles` isn't one of the specified types.

I should probably add a code comment here, but the idea of this logic is:

* When `styles` is requested, then all appropriate layout styles will be output anyway.
* When `block-layout-styles` is requested, then only those layout styles required to correctly render core block styles should be output — this gives us the desired fallback gap / layout support for classic themes for blocks like buttons and social links.